akka-http/src/main/scala/akka/http/scaladsl/server/Rejection.scala

```diff
@@ -92,6 +94,15 @@ final case class MalformedHeaderRejection(headerName: String, errorMsg: String,
   extends jserver.MalformedHeaderRejection with RejectionWithOptionalCause
 
 /**
+ * Rejection created by [[akka.http.scaladsl.server.directives.HeaderDirectives.checkSameOrigin]].
+ * Signals that the request was rejected because `Origin` header value is invalid.
+ */
+final case class InvalidOriginHeaderRejection(invalidOrigins: immutable.Seq[SHttpOrigin])
```

Comment:
Oh, I'm sorry - I missed this comment was left unadressed. 
I agree it would be a good idea to include the origins.